.github/workflows/scripts/pre-release/references.sh

```diff
@@ -32,3 +33,16 @@ if [[ "$results" != "" ]]; then
     exit 1
 fi
 
+# Verify documentation refers to the most recent release tag
+results=$(
+    find . -name "*.md" -print0 \
+    | xargs -0 grep -Pn "uses: slsa-framework/.*@(?!<|main|$RELEASE_TAG)" \
```

Comment:
do we currently allow @main for something? i know the internal actions reference main, but I would expect those only to be used in the .yml's not in the documentation

Reply:
`@main` is currently used in two documentation pages.

One is for the reuseable `detect-workflow` workflow. 

https://github.com/slsa-framework/slsa-github-generator/blob/9a62ed287f4ffc9ab9c45aa40264c92f10784f51/.github/actions/detect-workflow/README.md?plain=1#L27

The other is for the generic container builder, which doesn't have a tagged release yet.

https://github.com/slsa-framework/slsa-github-generator/blob/9a62ed287f4ffc9ab9c45aa40264c92f10784f51/internal/builders/container/README.md?plain=1#L70

https://github.com/slsa-framework/slsa-github-generator/blob/9a62ed287f4ffc9ab9c45aa40264c92f10784f51/internal/builders/container/README.md?plain=1#L143

I can shrink the search path to ignore the `.github` folder (and therefore the `detect-workflow` "infraction") if you wish.

But not much to be done about the container builder, I don't think. The search will need to handle it somehow, for now: either by allowing main or by omitting that directory as well.

I'd argue temporarily allowing `@main` is better simply because the threat (keeping an Action `@main` after a release exists) is far more obvious and therefore easier to detect, at which point the Action can be patched. Omitting the directory risks no one remembering it's omitted and therefore it remains with old tags for longer than it should.

I've added a comment stating `@main` should be forbidden once the container builder is released.

Reply:
I also feel really sorry for user `@main`, who must get a thousand notifications a day. 😂

Reply:
> `@main` is currently used in two documentation pages.
> 
> One is for the reuseable `detect-workflow` workflow.
> 
> https://github.com/slsa-framework/slsa-github-generator/blob/9a62ed287f4ffc9ab9c45aa40264c92f10784f51/.github/actions/detect-workflow/README.md?plain=1#L27

We should make this a more generic reusable workflow example like `uses: path/to/reusable_workflow.yml@v1.0`

> The other is for the generic container builder, which doesn't have a tagged release yet.

See my other comment.

Reply:
On `detect-workflow`, to clarify, are you suggesting creating another repo for this Action (and maybe others)? Would you suggest making this change before merging this PR?

Reply:
I don't think we need another repo. I'm just suggesting that we change the example reusable workflow used in the [.github/actions/detect-workflow/README.md](https://github.com/slsa-framework/slsa-github-generator/blob/9a62ed287f4ffc9ab9c45aa40264c92f10784f51/.github/actions/detect-workflow/README.md?plain=1#L27) from

```yaml
uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@main 
```

to

```yaml
uses: path/to/reusable_workflow.yml@v1.0
```

so that it doesn't need to be changed during releases. It's a generic action so the example in the doc doesn't need to use `slsa-framework/slsa-github-generator/.github/...`